content/sips/sip-354.md

```diff
@@ -0,0 +1,188 @@
+
+### Technical Specification
+
+<!--The technical specification should outline the public API of the changes proposed. That is, changes to any of the interfaces Synthetix currently exposes or the creations of new ones.-->
+
+Setters/Getters
+
+```
+function setInterestRateParameters(
+  uint128 lowUtilizationInterestRateGradient,
+  uint128 interestRateGradientBreakpoint,
+  uint128 highUtilizationInterestRateGradient
+) external;
+
+function getInterestRateParameters()
+  external
+  view
+  returns (
+    uint128 lowUtilizationInterestRateGradient,
+    uint128 interestRateGradientBreakpoint,
+    uint128 highUtilizationInterestRateGradient
+  );
+```
+
+Because utilization rate is dependent on delegated collateral from the LP side, there is no hook that forces the rate on the perps supermarket to be updated when someone (un)delegates. For this reason, a convenience method to manually update the interest rate that could be called the LP app has been added:
+
+```
+function updateInterestRate() external;
+```
+
+New views have been added to surface the current utliziationRate or the computed interestRate of the supermarket:
+
+```
+/**
+  * @notice Returns the current market interest rate
+  * @return rate
+  */
+function interestRate() external view returns (uint128 rate);
+
+/**
+  * @notice Returns the super market utilization rate
+  * @dev The rate is the minimumCredit / delegatedCollateral available.
+  * @dev Locked credit is the sum of all markets open interest * configured lockedOiRatio
+  * @dev delegatedCollateral is the credit capacity available to the supermarket
+  * @return rate
+  * @return delegatedCollateral
+  * @return lockedCredit credit locked based on OI & lockedOiRatio
+  */
+function utilizationRate()
+    external
+    view
+    returns (uint256 rate, uint256 delegatedCollateral, uint256 lockedCredit);
+```
+
+The integrators can get a trader's owed interest by calling the previously available `getOpenPosition` which now returns that value:
+
+```diff
+    function getOpenPosition(
+        uint128 accountId,
+        uint128 marketId
+    )
+        external
+        view
+        returns (
+          int256 totalPnl,
+          int256 accruedFunding,
+          int128 positionSize,
++         uint256 owedInterest
+        );
+```
+
+New events added:
+
+```
+/**
+  * @notice Gets emitted when the interest rate is updated.
+  */
+event InterestRateUpdated(uint128 indexed superMarketId, uint128 interestRate);
+
+/**
+  * @notice Emitted when interest rate parameters are set
+  */
+event InterestRateParametersSet(
+    uint256 lowUtilizationInterestRateGradient,
+    uint256 interestRateGradientBreakpoint,
+    uint256 highUtilizationInterestRateGradient
+);
+
+/**
+  * @notice Gets emitted after any trade settles and includes interest charged to the trader on previous position
+  */
+event InterestCharged(uint128 indexed accountId, uint256 interest);
```

Comment:
Instead of a new event, is it possible to add this to the current `OrderSettled` event if that is when this is charged? This should simplify things since all other changes to the position are included there, like accrued funding, size changes, pnl, fees, etc.

Reply:
@Tburm unfortunately we're hitting a stack too deep error due to the size of that event so had to break it up a bit.  i'm hoping there's ways to achieve what you are looking for with the indexed accountId but feel free to push back.  maybe we can break up `OrderSettled` into `OrderSettled1` and `OrderSettled2`, idk.  solidity man

Reply:
That makes sense, it's just a bit more overhead to handle these in separate events compared to having everything we need in `OrderSettled`. This solution with a single event for interest rates works, it will just need to be included in addition to `OrderSettled` to track positions so we need to make sure integrators and dashboards like Watcher are aware of the new event.